### PR TITLE
WIP: Add plugin for mail notification

### PR DESCRIPTION
**Pull request by Martchus**
This is just a first approach which only sends mails to me when any comment has been added. If you like to try it, please change the mail address.

A table to manage subscriptions would be required to go on. I'm currently trying to figure out how to do it.

See issue https://progress.opensuse.org/issues/10672

### CODE REVIEW

File: lib/OpenQA/ServerStartup.pm [hunk 1]
```diff
@@ -62,6 +62,10 @@ sub read_config {
         audit => {
             blacklist => '',
         },
+        mail_notification => {
+            from => undef,
+            base_url => 'http://localhost:9526/',
```
Comment (okurz) [inline lib/OpenQA/ServerStartup.pm]: should be changed

File: lib/OpenQA/WebAPI/Plugin/MailNotification.pm [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2015 SUSE LLC
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Plugin/MailNotification.pm]: update

File: lib/OpenQA/WebAPI/Plugin/MailNotification.pm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        if($comment->group_id) {
+            my $group = $app->db->resultset('JobGroups')->find($comment->group_id);
+            $subject = 'New comment in job group: ' . $group->name;
+            $path = '/group_overview/' . $group->id;
+        }
+        elsif($comment->job_id) {
+            my $job = $app->db->resultset('Jobs')->find($comment->job_id);
+            $subject = 'New comment in job group: ' . 'New comment in job: ' . $job->name;
+            $path = '/tests/' . $job->id . '#comments';
+        }
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Plugin/MailNotification.pm]: this branch should not make sense

File: lib/OpenQA/ServerStartup.pm (hunk 1 above)
Comment (aaannz) [inline lib/OpenQA/ServerStartup.pm]: There is already base_url in global section. Is this something different?

Comment (Martchus) [inline lib/OpenQA/ServerStartup.pm]: No, I will use the one in the global section.

File: lib/OpenQA/WebAPI/Plugin/MailNotification.pm [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+sub send_mail {
+    my ($to, $subject, $message) = @_;
+
+    my @header = (
+        To => $to,
+        Subject => $subject
+    );
+    if($config->{mail_notification}{from}) {
+        push(@header, From => $config->{mail_notification}{from});
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Plugin/MailNotification.pm]: shouldn't it refuse to send mail at all without from?

File: lib/OpenQA/WebAPI/Plugin/MailNotification.pm (hunk 3 above)
Comment (coolo) [inline lib/OpenQA/WebAPI/Plugin/MailNotification.pm]: you mean the else? Better save then sorry.

**Comment by Martchus** [comment]
This approach for mail notifications involves a database changes as well. Since we're currently focusing on the job groups, I will postpone this as it likely requires some more work anyways. I also can not reproduce the failing check locally.

**Comment by Martchus** [comment]
As we're following the fedmsg approach now, this is no longer required.

**Pull request closed without merging**